language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableCodeActionWithType.java

```diff
@@ -0,0 +1,284 @@
+                importsAcceptor.getNewImportTextEdits(), renamePositions);
+    }
+
+    /**
+     * Get all possible return type combinations for the type infer required method.
+     * Here if the type symbol is a union we will check it contains an error member.
+     * Other members of the union will then combine with the error type and produce the output.
+     * In here we will `any` type.
+     *
+     * @param typeSymbol Return type descriptor of the method.
+     * @param context    CodeActionContext
+     * @return {@link List<String>}
+     */
+    private List<String> getPossibleTypes(TypeSymbol typeSymbol, CodeActionContext context) {
+        typeSymbol = getRawType(typeSymbol, context);
+        Set<String> possibleTypes = new HashSet<>();
+        List<String> finalPossibleTypeSet = new ArrayList<>();
+        List<TypeSymbol> errorTypes = new ArrayList<>();
+        if (typeSymbol.typeKind() == TypeDescKind.UNION) {
+            ((UnionTypeSymbol) typeSymbol)
+                    .memberTypeDescriptors()
+                    .stream()
+                    .map(memberTypeSymbol -> getRawType(memberTypeSymbol, context))
+                    .forEach(memberTypeSymbol -> {
+                        if (memberTypeSymbol.typeKind() == TypeDescKind.UNION) {
+                            possibleTypes.addAll(
+                                    ((UnionTypeSymbol) memberTypeSymbol)
+                                            .memberTypeDescriptors()
+                                            .stream()
+                                            .map(memberTSymbol -> getRawType(memberTSymbol, context))
+                                            .map(symbol -> getTypeName(symbol, context))
+                                            .collect(Collectors.toList()));
+                        } else {
+                            if (memberTypeSymbol.typeKind() == TypeDescKind.ERROR
+                                    || CommonUtil.getRawType(memberTypeSymbol).typeKind() == TypeDescKind.ERROR) {
+                                errorTypes.add(memberTypeSymbol);
+                            } else {
+                                possibleTypes.add(getTypeName(memberTypeSymbol, context));
+                            }
+                        }
+                    });
+        } else {
+            String type = getTypeName(typeSymbol, context);
+            if (!type.equals("any")) {
+                return Collections.singletonList(type);
+            }
+        }
+
+        if (!errorTypes.isEmpty()) {
+            String errorTypeStr = errorTypes.stream()
+                    .map(type -> getTypeName(type, context))
+                    .collect(Collectors.joining("|"));
+            finalPossibleTypeSet.addAll(possibleTypes.stream()
+                    .filter(type -> !type.equals("any"))
+                    .map(type -> type + "|" + errorTypeStr).collect(Collectors.toSet()));
```

Comment:
we can have a early return here

Reply:
We cannot right because `addAll` method will return a boolean value.